### DISCUSSION TITLE
Macos-14 runner Image 20240422.3 serious performance degradation

### DISCUSSION THREAD

**Discussion by skeeping** [actions/virtual-environments#9751] [General questions]
### Description

Build times went from 10-11 minutes to 16-21 minutes when they switched from 20240415.6 to 20240422.3.
No other things have changed and i reran a build that ran on 20240415.6 which reran on 20240422.3 and the build time went from [10m 47s] to [19m 6s]

### Platforms affected

- [ ] Azure DevOps
- [X] GitHub Actions - Standard Runners
- [x] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 11
- [ ] macOS 12
- [ ] macOS 13
- [ ] macOS 13 Arm64
- [ ] macOS 14
- [X] macOS 14 Arm64
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

20240422.3

### Is it regression?

yes was good in 20240415.6 

### Expected behavior

build times should not increase build 60-100% but should remain at 10-11 minutes

### Actual behavior

build times are between 16 and 21 minutes

### Repro steps

Build an ios application with 20240415.6 note the time, Rebuild the same application with 20240422.3 and compare and contrast

**Comment by Alexey-Ayupov** [comment]
The issue was converted to the discussions due to Increase visibility of the problem.
It looks like performance issue is on the new XCode simulators version said and not on the runners itself.

**Reply by skeeping** [reply]
the main type of build time shouldn't have had anything to do with the simulator as it was just a normal compile and it should be the same version of xcode 15.2 that we are using.

**Reply by skeeping** [reply]
updating this because i have now tried on a larger runner with the same performance hit

**Reply by PatTheMav** [reply]
FWIW we see the same - when macOS 14 runners where introduced, average build times decreased heavily compared to macOS 13 runners. Building the project without caching took about 4.5 minutes and about 2.5 minutes with caching via `ccache` and a 99% cache hit rate.

Currently we can observe build times of about 12 minutes, sometimes even 24 minutes, for an uncached build and about 5 minutes with a 99% `ccache` hit rate.

We do not use the simulator in a fashion, we create an Xcode project via `CMake` and execute it via `xcodebuild`.
